Build failure: pkgsCross.mingwW64.libxml2 and pkgsCross.mingw32.libxml2

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

`nix build .#pkgsCross.mingwW64.libxml2`

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/b9ky2dbk2cp672ig8ankp8gl5ski2vra-libxml2-x86_64-w64-mingw32-2.14.3.drv' failed with exit code 1;
       last 25 log lines:
       > checking for sys/types.h... yes
       > checking for unistd.h... yes
       > checking for dlfcn.h... no
       > checking for objdir... .libs
       > checking if x86_64-w64-mingw32-gcc supports -fno-rtti -fno-exceptions... no
       > checking for x86_64-w64-mingw32-gcc option to produce PIC... -DDLL_EXPORT -DPIC
       > checking if x86_64-w64-mingw32-gcc PIC flag -DDLL_EXPORT -DPIC works... yes
       > checking if x86_64-w64-mingw32-gcc static flag -static works... yes
       > checking if x86_64-w64-mingw32-gcc supports -c -o file.o... yes
       > checking if x86_64-w64-mingw32-gcc supports -c -o file.o... (cached) yes
       > checking whether the x86_64-w64-mingw32-gcc linker (x86_64-w64-mingw32-ld) supports shared libraries... yes
       > checking dynamic linker characteristics... Win32 ld.exe
       > checking how to hardcode library paths into programs... immediate
       > checking whether stripping libraries is possible... yes
       > checking if libtool supports shared libraries... yes
       > checking whether to build shared libraries... no
       > checking whether to build static libraries... yes
       > checking for __attribute__((destructor))... yes
       > checking for stdint.h... (cached) yes
       > checking for x86_64-w64-mingw32-gcc options needed to detect all undeclared functions... none needed
       > checking whether getentropy is declared... no
       > checking whether glob is declared... no
       > checking whether mmap is declared... no
       > checking for libiconv... no
       > configure: error: libiconv not found
       For full logs, run:
         nix log /nix/store/b9ky2dbk2cp672ig8ankp8gl5ski2vra-libxml2-x86_64-w64-mingw32-2.14.3.drv
```

### Additional context

_No response_

### System metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.34, NixOS, 25.11 (Xantusia), 25.11.20250621.4206c4c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/si3k775n94ax09w3cjgc5hr2vjs84v7m-source`
```

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/libxml2/default.nix
+++ b/pkgs/development/libraries/libxml2/default.nix
@@ -82,7 +82,7 @@ stdenv.mkDerivation (finalAttrs: {
     [
       findXMLCatalogs
     ]
-    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    ++ lib.optionals (stdenv.hostPlatform.isDarwin || stdenv.hostPlatform.isMinGW) [
       libiconv
     ]
     ++ lib.optionals icuSupport [

